src/test/java/com/google/sps/unit/ServletUtilsTest.java

```diff
@@ -0,0 +1,113 @@
+import org.mockito.junit.MockitoRule;
+
+public class ServletUtilsTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(), new LocalURLFetchServiceTestConfig());
+
+  @Mock HttpServletRequest httpServletRequest;
+  @Mock HttpServletResponse httpServletResponse;
+  @Mock PrintWriter printWriter;
+
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  // Test ServletUtils getRquestParam() function gets parameter in Optional container correctly.
+  @Test
+  public void testHTTPRequestGetValidParameter() throws Exception {
+    when(httpServletRequest.getParameter("myKey")).thenReturn("myValue");
+
+    Optional<String> parameter =
+        ServletUtils.getRequestParam(httpServletRequest, httpServletResponse, "myKey");
+    Assert.assertTrue(parameter.isPresent());
+    Assert.assertEquals(parameter.get(), "myValue");
+  }
+
+  // Test ServletUtils getRquestParam() function correctly returns empty Optional container when
+  // parameter key does not exist.
+  @Test
+  public void testHTTPRequestGetInvalidParameter() throws Exception {
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    Optional<String> parameter =
+        ServletUtils.getRequestParam(httpServletRequest, httpServletResponse, "anInvalidKey");
+    Assert.assertTrue(!parameter.isPresent());
+    verify(printWriter).println("No anInvalidKey in the query URL.");
+  }
+
+  // Test ServletUtils getRquestParam() function correctly returns empty Optional container when
+  // provided parameter key is null.
+  @Test
+  public void testHTTPRequestGetNullParameter() throws Exception {
+    when(httpServletResponse.getWriter()).thenReturn(printWriter);
+
+    Optional<String> parameter =
+        ServletUtils.getRequestParam(httpServletRequest, httpServletResponse, null);
+    Assert.assertTrue(!parameter.isPresent());
+    verify(printWriter).println("No null in the query URL.");
+  }
+
+  // Test finding an Election Entity in Datastore
+  @Test
+  public void testFindExistingElectionInDatastore() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    Entity electionEntity = new Entity("Election");
+    electionEntity.setProperty("id", "9999");
+    electionEntity.setProperty("name", "myElection");
+    electionEntity.setProperty("scope", "myScope");
+    electionEntity.setProperty("date", "myDate");
+    electionEntity.setProperty("contests", new HashSet<Long>());
+    electionEntity.setProperty("propositions", new HashSet<Long>());
+    ds.put(electionEntity);
+
+    Optional<Entity> foundEntity = ServletUtils.findElectionInDatastore(ds, "9999");
+    Assert.assertTrue(foundEntity.isPresent());
+    Assert.assertEquals(foundEntity.get().getProperty("name"), "myElection");
```

Comment:
Just curious, would assertEquals(foundEntity.get(), electionEntity) fail here?

Reply:
That works 😅